crates/fluvio-connector-package/src/validation.rs

```diff
@@ -0,0 +1,247 @@
+use anyhow::anyhow;
+
+use crate::{
+    config::ConnectorConfig,
+    metadata::{ConnectorMetadata, Direction, Deployment, Secrets, Parameters},
+};
+
+pub fn validate_config(
+    metadata: &ConnectorMetadata,
+    config: &ConnectorConfig,
+) -> anyhow::Result<()> {
+    validate_direction(&metadata.direction, config)?;
+    validate_deployment(&metadata.deployment, config)?;
+    validate_secrets(&metadata.secrets, config)?;
+    validate_parameters(&metadata.parameters, config)?;
+    Ok(())
+}
+
+fn validate_direction(direction: &Direction, config: &ConnectorConfig) -> anyhow::Result<()> {
+    let meta_is_source = direction.is_source();
+    let cfg_is_source = config.type_.ends_with("source");
+    if meta_is_source != cfg_is_source {
+        let (meta_dir, cfg_dir) = if meta_is_source {
+            ("source", "sink")
+        } else {
+            ("dest", "source")
+        };
+        return Err(anyhow!(
+            "direction in metadata: '{}' does not correspond direction in config: '{}'",
+            meta_dir,
+            cfg_dir
+        ));
+    }
+    Ok(())
+}
+
+fn validate_deployment(deployment: &Deployment, config: &ConnectorConfig) -> anyhow::Result<()> {
+    let image = format!(
+        "infinyon/fluvio-connect-{}:{}",
```

Comment:
change this to constant

Reply:
also moved as a constant to ConnectorConfig